pkg/kubectl/cmd/util/factory.go

```diff
@@ -145,6 +145,9 @@ type Factory struct {
 	// UpdatePodSpecForObject will call the provided function on the pod spec this object supports,
 	// return false if no pod spec is supported, or return an error.
 	UpdatePodSpecForObject func(obj runtime.Object, fn func(*api.PodSpec) error) (bool, error)
+	// UpdateSelectorForObject calls fn with that applicable selector type to update.
+	UpdateSelectorForObject func(obj runtime.Object, fn func(*map[string]string, *unversioned.LabelSelector) error) error
```

Comment:
It would be better for your fn to return the new selector or an error, rather than passing by reference.  Returning the new value for "value types" is preferred generally.